adapters/http4s/src/main/scala/caliban/Http4sAdapter.scala

```diff
@@ -366,14 +366,15 @@ object Http4sAdapter {
             for {
               msg    <- RIO.fromEither(decode[Json](text))
               msgType = msg.hcursor.downField("type").success.flatMap(_.value.asString).getOrElse("")
-              _      <- RIO.whenCase(msgType) {
+              _      <- RIO.whenCase[R, String](msgType) {
```

Comment:
I guess it didn't compile in Scala 3 because explicit types were missing.

Reply:
Yeah weirdly this one seems necessary with Scala 3.